Tasks/Common/RemoteDeployer/SessionHelper.ps1

```diff
@@ -0,0 +1,139 @@
+function ConnectTo-RemoteMachine {
+    [CmdletBinding()]
+    Param (
+        [string] $computerName,
+        [string] $port,
+        [pscredential] $credential,
+        [string] $authentication,
+        [string] $sessionName,
+        [string] $sessionConfigurationName,
+        [switch] $useSsl,
+        [ValidateRange(2,10)]
+        [int] $maxRetryLimit = 3
+    )
+
+    Trace-VstsEnteringInvocation $MyInvocation
+    try {
+        $retryCount = 0;
+        $isConnectionComplete = $false
+        $newPsSessionCommand = Get-NewPSSessionCommand -computerName $computerName `
+                                                       -port $port `
+                                                       -authentication $authentication `
+                                                       -sessionName $sessionName `
+                                                       -useSsl:$useSsl `
+                                                       -sessionConfigurationName $sessionConfigurationName `
+                                                       -NoCredential:($credential -eq $null)
+
+        while ($retryCount -lt $maxRetryLimit) {
+            Write-Verbose "Trying to establish connection: Attempt #$($retryCount + 1)"
+            $session = (Invoke-Expression $newPsSessionCommand)
```

Comment:
no action required - we may add try catch for session creation Invoke-Expression

This can be done later once we have data for this